src/app/core/access-checker/access-checker.service.spec.js

```diff
@@ -130,11 +130,11 @@ describe('Access Checker Service:', () => {
 		it('should not update the cache when the access checker provider fails', () => {
 			return accessChecker.get('provideronly').then((result) => {
 				should.fail('Fail provider should throw an error');
-			}, (err) => {
+			}).catch((err) => {
 				// Should have errored
 				should.exist(err);
 
-				return CacheEntry.findOne({ key: 'provideronly' }).exec()
```

Comment:
You removed the `exec()` calls in various places.  Mongoose by default will return a thenable, but not a true Promise.  `exec()` returns a true Promise and would be required in cases where we want to call `catch` on the result.  It may also be required when you want to use `Promise.all(...)`.  Mongoose also [recommends](https://mongoosejs.com/docs/promises.html#should-you-use-exec-with-await?) when using `async/await`

Reply:
@jrassa sounds great! Do we want to make this PR a big move to `async/await` ?

Reply:
@jrassa for now, I restored the `.exec()` calls and added them to a few places.